### ISSUE TITLE
Reusing database service object

### ISSUE THREAD

**Issue by Wyzix33** [box/t3js#190]
Hi, i want to use Lokijs inside a service and want to reuse database object in some modules so i did this:

```
Box.Application.addService('db', function (application) {
	function db() {
		this.db = new loki('psi', {env: 'BROWSER'});
		this.users = this.db.addCollection('pagini',{ unique: ['id'] });
                this.load('url');
	}
	db.prototype = {
                load: function (url) {
               //loading database from server and insert to lokijs
                },
		search: function (q) {
			//perform query and return results 
		}
	}
	var database = new db();
	return {
		create: function(options) {
		   return database;
		}
	};
});
```
i have a menu module that is loaded first that calls `var db = context.getService('db').create();` and i can call `db.search(....);`
and in all my modules i have to do the same thing:

```
var db = context.getService('db').create();
var res = db.search(....);
```
and it works,... but somehow it seems wrong, 

what is the best method to do this? maybe something like this, but also it seems a bit overkill:
```
var database;
	return {
		create: function(options) {
		   database  = new db();
		   return database;
		},
		database: function() {
			return database;
		}
	};
```
and i can call `context.getService('db').create();` first time in my menu module and `context.getService('db').database();` in others ... 
Please help ...
P.S - Is it a bad practice to have multiple services, one for each functionality?  or should i just make one service and add all other methods inside return?
In other words is it bad to have

```
var service = context.getService('service1');
var otherservice = context.getService('service2');
var someotherservice = context.getService('service3');
```
or just 
`var service = context.getService('service1');`
and then use 
```
service.dosomething();
service.dosomethingelese();
```
any performance loss one way or the other? 

Thanks

**Comment by j3tan** [comment]
Preferably you just call getService once since, but it doesn't matter since services are basically singletons so you'll only create the object once.

preferably you do something like:
```
var dbService = context.getService('db');
var db = dbService.create();
```
